scrapy/core/http2/protocol.py

```diff
@@ -0,0 +1,204 @@
+            self._protocol_error = e
+
+            # We lose the transport connection here
+            self.transport.loseConnection()
+        finally:
+            self._write_to_transport()
+
+    def connectionLost(self, reason=connectionDone):
+        """Called by Twisted when the transport connection is lost.
+        No need to write anything to transport here.
+        """
+        # Pop all streams which were pending and were not yet started
+        # NOTE: Stream.close() pops the element from the streams dictionary
+        # which raises `RuntimeError: dictionary changed size during iteration`
+        # Hence, we copy the streams into a list.
+        for stream in list(self.streams.values()):
+            stream.close(StreamCloseReason.CONNECTION_LOST, self._protocol_error)
+
+        self.conn.close_connection()
+
+        if not reason.check(connectionDone):
+            logger.warning("Connection lost with reason " + str(reason))
+
+    def _handle_events(self, events):
+        """Private method which acts as a bridge between the events
+        received from the HTTP/2 data and IH2EventsHandler
+
+        Arguments:
+            events {list} -- A list of events that the remote peer
+                triggered by sending data
+        """
+        for event in events:
+            logger.debug(event)
+            if isinstance(event, DataReceived):
+                self.data_received(event)
+            elif isinstance(event, ResponseReceived):
+                self.response_received(event)
+            elif isinstance(event, StreamEnded):
+                self.stream_ended(event)
+            elif isinstance(event, StreamReset):
+                self.stream_reset(event)
+            elif isinstance(event, WindowUpdated):
+                self.window_updated(event)
+            elif isinstance(event, SettingsAcknowledged):
+                self.settings_acknowledged(event)
+            else:
+                logger.info("Received unhandled event {}".format(event))
+
+    # Event handler functions starts here
+    def data_received(self, event: DataReceived):
+        self.streams[event.stream_id].receive_data(event.data, event.flow_controlled_length)
+
+    def response_received(self, event: ResponseReceived):
+        self.streams[event.stream_id].receive_headers(event.headers)
+
+    def settings_acknowledged(self, event: SettingsAcknowledged):
+        # Send off all the pending requests
+        # as now we have established a proper HTTP/2 connection
+        self._send_pending_requests()
+
+    def stream_ended(self, event: StreamEnded):
+        self.streams[event.stream_id].close(StreamCloseReason.ENDED)
+
+    def stream_reset(self, event: StreamReset):
+        self.streams[event.stream_id].close(StreamCloseReason.RESET)
+
+    def window_updated(self, event: WindowUpdated):
+        stream_id = event.stream_id
+        if stream_id != 0:
+            self.streams[stream_id].receive_window_update()
```

Comment:
```suggestion
        if event.stream_id != 0:
            self.streams[event.stream_id].receive_window_update()
```

A small simplification.